mm2src/coins/z_coin/z_rpc.rs

```diff
@@ -476,10 +478,12 @@ impl ZRpcOps for NativeClient {
     async fn check_tx_existence(&self, tx_id: TxId) -> bool {
         let mut attempts = 0;
         loop {
-            match self.get_raw_transaction_bytes(&H256Json::from(tx_id.0)).compat().await {
+            let tx_hash = H256Json::from(tx_id.0).reversed();
+            let tx = self.get_raw_transaction_bytes(&tx_hash).compat().await;
+            match tx {
                 Ok(_) => break,
                 Err(e) => {
-                    error!("Error on getting tx {}", tx_id);
+                    error!("Error on getting tx {}: err: {e:?}", tx_id);
                     if e.to_string().contains(NO_TX_ERROR_CODE) {
```

Comment:
I suggest not to check specifically for the NO_TX_ERROR_CODE error but return after 3 attempts for any error Now for errors other than NO_TX_ERROR_CODE we would loop endlessly.
Same for check_tx_existence impl for the light client.

(btw 3 retries x 30sec may not always be enough for a transaction to be available for get_raw_transaction, because transactions may be not mined yet but also may be missed in a node's mempool)

Reply:
increased this to 5 tries